internal/alias/target/repository_alias.go

```diff
@@ -0,0 +1,264 @@
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "nil embedded Alias")
+	case a.Value == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "no value")
+	case a.ScopeId == "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "no scope id")
+	case a.PublicId != "":
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "public id not empty")
+	}
+	a = a.clone()
+
+	id, err := newAliasId(ctx)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	a.PublicId = id
+
+	oplogWrapper, err := r.kms.GetWrapper(ctx, a.ScopeId, kms.KeyPurposeOplog)
+	if err != nil {
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get oplog wrapper"))
+	}
+
+	metadata := newAliasMetadata(a, oplog.OpType_OP_TYPE_CREATE)
+
+	var newAlias *Alias
+	_, err = r.writer.DoTx(
+		ctx,
+		db.StdRetryCnt,
+		db.ExpBackoff{},
+		func(_ db.Reader, w db.Writer) error {
+			newAlias = a.clone()
+			err := w.Create(
+				ctx,
+				newAlias,
+				db.WithOplog(oplogWrapper, metadata),
+			)
+			if err != nil {
+				return errors.Wrap(ctx, err, op)
+			}
+			return nil
+		},
+	)
+
+	if err != nil {
+		if errors.IsUniqueError(err) {
+			switch {
+			case strings.Contains(err.Error(), `"alias_value_uq"`):
+				return nil, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("alias value %q is already in use", a.Value)))
+			case strings.Contains(err.Error(), `"alias_target_scope_id_name_uq"`):
+				return nil, errors.Wrap(ctx, err, op, errors.WithMsg(fmt.Sprintf("in scope %q, the name %q is already in use", a.ScopeId, a.Name)))
+			}
+		}
+		if strings.Contains(err.Error(), `violates foreign key constraint "target_fkey"`) {
+			return nil, errors.Wrap(ctx, err, op, errors.WithCode(errors.NotFound), errors.WithMsg("target with specified destination id %q was not found", a.GetDestinationId()))
+		}
+		return nil, errors.Wrap(ctx, err, op)
+	}
+	return newAlias, nil
+}
+
+// UpdateAlias updates the repository entry for a.PublicId with the
+// values in a for the fields listed in fieldMask. It returns a new
+// Alias containing the updated values and a count of the number of
+// records updated. a is not changed.
+func (r *Repository) UpdateAlias(ctx context.Context, a *Alias, version uint32, fieldMask []string, opt ...Option) (*Alias, int, error) {
+	const op = "target.(Repository).UpdateAlias"
+	switch {
+	case a == nil:
+		return nil, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "nil Alias")
+	case a.Alias == nil:
+		return nil, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "nil embedded Alias")
+	case a.PublicId == "":
+		return nil, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "no public id")
+	case len(fieldMask) == 0:
+		return nil, db.NoRowsAffected, errors.New(ctx, errors.EmptyFieldMask, op, "empty field mask")
+	case version == 0:
+		return nil, db.NoRowsAffected, errors.New(ctx, errors.InvalidParameter, op, "no version")
+	}
+
+	var dbMask, nullFields []string
```

Comment:
Is there a reason not to use the dbw.BuildUpdatePaths pattern here?(ex: https://github.com/hashicorp/boundary/blob/main/internal/credential/static/repository_credential.go#L384-L402)